### PR TITLE
k3s: token file

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/cluster/k3s/default.nix [hunk 1]
```diff
@@ -35,10 +35,20 @@ in
 
     token = mkOption {
       type = types.str;
```
Comment (SuperSandro2000) [inline nixos/modules/services/cluster/k3s/default.nix:37]:
```suggestion
      type = types.nullOr types.str;
```

Comment (Mic92) [reply nixos/modules/services/cluster/k3s/default.nix:37]: This is too late. It already defaults too an empty string.

Comment (Mic92) [reply nixos/modules/services/cluster/k3s/default.nix:37]: If fixed it for tokenFile so.

**Review by mohe2015 — APPROVED** [review]
LGTM!

**Pull request merged**